api/src/middlewares/services/EmailService/NotifyService.ts

```diff
@@ -0,0 +1,115 @@
+import { NotifyClient, RequestError, SendEmailResponse } from "notifications-node-client";
+import config from "config";
+import pino, { Logger } from "pino";
+import { ApplicationError } from "../../../ApplicationError";
+import * as additionalContexts from "./additionalContexts.json";
+import { EmailServiceProvider, NotifySendEmailArgs } from "./types";
+import * as templates from "./templates";
+import { FormField } from "../../../types/FormField";
+import { answersHashMap } from "../helpers";
+import { AnswersHashMap } from "../../../types/AnswersHashMap";
+
+const previousMarriageDocs = {
+  Divorced: "decree absolute",
+  "Dissolved civil partner": "final order",
+  Widowed: "late partner's death certificate",
+  "Surviving civil partner": "late partner's death certificate",
+  Annulled: "decree of nullity",
+};
+export class NotifyService implements EmailServiceProvider {
+  notify: NotifyClient;
+  logger: Logger;
+  templates: Record<string, string>;
+  constructor() {
+    const apiKey = config.get("notifyApiKey");
+    const standardTemplate = config.get("notifyTemplateStandard");
+    if (!apiKey) {
+      throw new ApplicationError("NOTIFY", "NO_API_KEY", 500);
+    }
+    if (!standardTemplate) {
+      throw new ApplicationError("NOTIFY", "NO_TEMPLATE", 500);
+    }
+    this.templates = {
+      standard: standardTemplate as string,
+    };
+    this.notify = new NotifyClient(apiKey as string);
+    this.logger = pino().child({ service: "Notify" });
+  }
+
+  async send(fields: FormField[], template: string, reference: string) {
+    const emailArgs = this.buildSendEmailArgs(fields, template, reference);
+    return this.sendEmail(emailArgs, reference);
+  }
+
+  async sendEmail({ template, emailAddress, options }: NotifySendEmailArgs, reference: string) {
+    try {
+      const response = await this.notify.sendEmail(template, emailAddress, options);
+      this.logger.info(`Reference ${reference} user email sent successfully with Notify id: ${(response.data as SendEmailResponse).id}`);
+      return response.data as SendEmailResponse;
```

Comment:
inlining type casts end u getting quite unreadable. if possible, fix the type so you don't need to do so, or abstract the cast out into it's own declaration.

```suggestion
      const response = await this.notify.sendEmail(template, emailAddress, options);
      const data = response.data as SendEmailResponse;
      this.logger.info(`Reference ${reference} user email sent successfully with Notify id: ${data.id}`);
      return data;
```